### ISSUE TITLE
slot_loc and global_id may be uninitialised

### ISSUE THREAD

**Issue by radarsat1** [libmapper/libmapper#20]
I'm getting a couple of "may be uninitialised" warnings in device.c with gcc 6.1, but I'm not familiar enough with the code to tell at first glance if the warning is bad or not.  It seems at least `slot_loc` is only initialized when `index_count >= 0`, but the code may continue either way.  Can you verify Joe?

~~~
device.c:603:50: warning: ‘slot_loc’ may be used uninitialized in this function [-Wmaybe-uninitialized]
device.c:587:34: warning: ‘global_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
~~~

I only get a couple of warnings like that, otherwise the code compiles perfectly, so we may as well get rid of them before release.

**Comment by malloch** [comment]
I think I have resolved these 2 warnings in the last commit.

**Comment by radarsat1** [comment]
Fixed in 562372edd8f4f89e7102f0ad70c064047055f738
